### PR TITLE
Infinite scroll on pickers

### PR DESCRIPTION
**Pull request by amrutha-rajiv**
##### **Description**
 - added onScrollDown callback in BasePickerResults - triggered while scrolling

##### **Reviewers**
* @jgodi
* @more

### CODE REVIEW

File: src/elements/picker/extras/picker-results/PickerResults.ts [hunk 1]
```diff
@@ -260,16 +279,30 @@ export class BasePickerResults {
...
+    }
 }
 
 @Component({
     selector: 'picker-results',
     host: {
         'class': 'active',
-        '[hidden]': 'matches.length === 0'
+        '[hidden]': 'matches.length === 0',
+        '(scroll)': 'onScrollDown($event)'
```
Comment (jgodi) [inline src/elements/picker/extras/picker-results/PickerResults.ts]: Can you move this to `@HostListener`? These will be deprecated eventually.

https://github.com/angular/angular/blob/4.0.0-beta.2/modules/%40angular/core/src/metadata/directives.ts#L981

File: src/elements/picker/extras/entity-picker-results/EntityPickerResults.ts [hunk 2]
```diff
@@ -65,10 +66,12 @@ import { NovoLabelService } from '../../../../services/novo-label-service';
                     </p>
                 </item-content>
             </novo-list-item>
+            <novo-loading theme="line" *ngIf="isLoading && matches.length > 0"></novo-loading>            
         </novo-list>
     `,
     host: {
-        '[hidden]': 'matches.length === 0'
+        '[hidden]': 'matches.length === 0',
+        '(scroll)': 'onScrollDown($event)'
```
Comment (jgodi) [inline src/elements/picker/extras/entity-picker-results/EntityPickerResults.ts]: Can you move this to `@HostListener`? These will be deprecated eventually.

https://github.com/angular/angular/blob/4.0.0-beta.2/modules/%40angular/core/src/metadata/directives.ts#L981

File: src/elements/picker/extras/entity-picker-results/EntityPickerResults.ts [hunk 3]
```diff
@@ -8,7 +8,8 @@ import { NovoLabelService } from '../../../../services/novo-label-service';
 @Component({
     selector: 'entity-picker-results',
     template: `
-        <novo-list *ngIf="matches.length > 0" direction="vertical">
+        <novo-list *ngIf="matches.length > 0" direction="vertical"
+             (scroll)="onScrollDown($event)">
```
Comment (jgodi) [inline src/elements/picker/extras/entity-picker-results/EntityPickerResults.ts]: Do you need both the host scroll and this one?

File: src/elements/picker/extras/picker-results/PickerResults.ts (hunk 1 above)
Comment (jgodi) [inline src/elements/picker/extras/picker-results/PickerResults.ts]: The cool thing with it is that you can do:

```
@HostListener('click', ['$event.target'])
onClick(target) {
    // Already have the target
}
```

File: src/elements/picker/extras/entity-picker-results/EntityPickerResults.ts (hunk 3 above)
Comment (amrutha-rajiv) [inline src/elements/picker/extras/entity-picker-results/EntityPickerResults.ts]: nope, removing.

File: src/elements/picker/extras/picker-results/PickerResults.ts (hunk 1 above)
Comment (amrutha-rajiv) [inline src/elements/picker/extras/picker-results/PickerResults.ts]: Moving

**Pull request merged**